### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by dchengTSC**

### CODE REVIEW

File: src/lib_json/json_value.cpp [hunk 1]
```diff
@@ -718,7 +718,7 @@ bool Value::asBool() const {
   case uintValue:
     return value_.uint_ ? true : false;
   case realValue:
-    return value_.real_ ? true : false;
+    return static_cast<bool>(value_.real_) ? true : false;
```
Comment (stephenkwagner) [inline src/lib_json/json_value.cpp:721]: the ternary here is redundant, but I guess that is how they wrote it.

File: src/lib_json/json_value.cpp [hunk 2]
```diff
@@ -1049,8 +1049,8 @@ bool Value::removeIndex(ArrayIndex index, Value* removed) {
   ArrayIndex oldSize = size();
   // shift left all items left, into the place of the "removed"
   for (ArrayIndex i = index; i < (oldSize - 1); ++i){
-    CZString key(i);
-    (*value_.map_)[key] = (*this)[i + 1];
+    CZString indexKey(i);
```
Comment (stephenkwagner) [inline src/lib_json/json_value.cpp:1052]: did they change `key` to `indexKey`??

Comment (dchengTSC) [reply src/lib_json/json_value.cpp:1052]: Nope, I just renamed it from `key` to `indexKey` so it wouldn't shadow the other `key` variable.

File: src/lib_json/json_value.cpp (hunk 1 above)
Comment (dchengTSC) [reply src/lib_json/json_value.cpp:721]: Yeah :/ I originally turned the specific warnings off for this specific file, but then decided to just fix them.

File: src/lib_json/json_value.cpp (hunk 2 above)
Comment (stephenkwagner) [reply src/lib_json/json_value.cpp:1052]: okay.  it was not obvious why this change was needed.

**Comment by stephenkwagner** [comment]
👍

**Pull request merged**